src/analyses/variable-sensitivity/constant_array_abstract_object.cpp

```diff
@@ -348,11 +353,11 @@ sharing_ptrt<array_abstract_objectt>
       {
         if(is_top())
         {
-          copy->clear_top();
+          result->clear_top();
         }
 
-        copy->map[index_value]=value;
-        return copy;
+        result->map[index_value]=value;
```

Comment:
Here (and in lines 396, 409) we could check if the new value differs from the old one before writing to the map, and use `find(..., tvt(true))` or `insert(..., tvt(false))` to modify/insert an element into the map if we know for certain that it's in there or not.